### PR TITLE
[FEATURE] Add raw attention scores to the AttentionCell #951

### PR DESCRIPTION
**Pull request by emilmont**
## Description ##
For implementing a pointer mechanism in sequence to sequence models it is very practical to re-use attention cells. For example see the Attention-Based Copy Mechanism described in Jia, Robin, and Percy Liang. "Data recombination for neural semantic parsing." arXiv preprint arXiv:1606.03622 (2016).
The proposal is to additionally return the raw attention scores in the AttentionCell.

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

## Comments ##
- The change is a backward compatible

cc @dmlc/gluon-nlp-team

### CODE REVIEW

**Comment by mli** [comment]
Job PR-964/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-964/1/index.html

**Comment by mli** [comment]
Job PR-964/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-964/2/index.html

File: src/gluonnlp/model/attention_cell.py [hunk 1]
```diff
@@ -148,6 +159,8 @@ def __call__(self, query, key, value=None, mask=None):  # pylint: disable=argume
             Shape (batch_size, query_length, context_vec_dim)
         att_weights : Symbol or NDArray
             Attention weights. Shape (batch_size, query_length, memory_length)
+        [att_scores]: Symbol or NDArray
```
Comment (sxjscience) [inline src/gluonnlp/model/attention_cell.py]: No `[]`

Comment (emilmont) [reply src/gluonnlp/model/attention_cell.py]: OK, I removed the brackets.

**Comment by mli** [comment]
Job PR-964/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-964/5/index.html

**Pull request merged**